src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/GlobalSearchService.kt

```diff
@@ -0,0 +1,195 @@
+  private val telemetryClient: TelemetryClient,
+  private val authenticationHolder: AuthenticationHolder
+) {
+  companion object {
+    private val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun findByGlobalSearchCriteria(globalSearchCriteria: GlobalSearchCriteria, pageable: Pageable): Page<Prisoner> {
+    validateSearchForm(globalSearchCriteria)
+    if (globalSearchCriteria.offenderNo != null) {
+      queryBy(globalSearchCriteria,pageable) { idMatch(it) } onMatch {
+        customEventForFindBySearchCriteria(globalSearchCriteria, it.matches.size)
+        return PageImpl(it.matches, pageable, it.totalHits)
+      }
+    }
+    if (!(globalSearchCriteria.firstName.isNullOrBlank() && globalSearchCriteria.lastName.isNullOrBlank())) {
+      if (globalSearchCriteria.includeAliases) {
+        queryBy(globalSearchCriteria,pageable) { nameMatchWithAliases(it) } onMatch {
+          customEventForFindBySearchCriteria(globalSearchCriteria, it.matches.size)
+          val ret = PageImpl(it.matches, pageable, it.totalHits)
+          return ret
+        }
+      } else {
+        queryBy(globalSearchCriteria,pageable) { nameMatch(it) } onMatch {
+          customEventForFindBySearchCriteria(globalSearchCriteria, it.matches.size)
+          return PageImpl(it.matches, pageable, it.totalHits)
+        }
+      }
+    }
+    return return PageImpl(listOf(), pageable, 0L)
+  }
+
+  private fun validateSearchForm(globalSearchCriteria: GlobalSearchCriteria) {
+    if (!globalSearchCriteria.isValid()) {
+      log.warn("Invalid search  - no criteria provided")
+      throw BadRequestException("Invalid search  - please provide at least 1 search parameter")
+    }
+  }
+
+  private fun queryBy(
+    globalSearchCriteria: GlobalSearchCriteria,
+    pageable: Pageable,
+    queryBuilder: (globalSearchCriteria: GlobalSearchCriteria) -> BoolQueryBuilder?
+  ): GlobalResult {
+    val query = queryBuilder(globalSearchCriteria)
+    return query?.let {
+      val searchSourceBuilder = SearchSourceBuilder().apply {
+        query(query.withDefaults(globalSearchCriteria))
+        size(pageable.pageSize)
+        from(pageable.offset.toInt())
+        sort("_score")
+        sort("prisonerNumber")
+      }
+      val searchRequest = SearchRequest(arrayOf(getIndex()), searchSourceBuilder)
+      val searchResults = searchClient.search(searchRequest)
+      val prisonerMatches = getSearchResult(searchResults)
+      return if (prisonerMatches.isEmpty()) GlobalResult.NoMatch else GlobalResult.Match(prisonerMatches,searchResults.hits.totalHits?.value ?: 0)
+    } ?: GlobalResult.NoMatch
+  }
+
+  private fun idMatch(globalSearchCriteria: GlobalSearchCriteria): BoolQueryBuilder? {
+    with(globalSearchCriteria) {
+      return QueryBuilders.boolQuery()
+        .mustMultiMatchKeyword(
+          offenderNo?.canonicalPNCNumber(),
+          "prisonerNumber",
+          "bookingId",
+          "pncNumber",
+          "croNumber",
```

Comment:
you might need to check croNumber search works as expected, since cro has a slash, it has to be keyword, which means it ends up being case sensitive

Reply:
Think its marked as a keyword already